packages/toolpad-app/src/appDom/index.ts

```diff
@@ -1063,8 +1063,10 @@ function createRenderTreeNode(node: AppDomNode): RenderTreeNode | null {
   }
 
   if (isQuery(node) || isMutation(node)) {
+    // This is hacky, should we delegate this check to the datasources?
```

Comment:
It would probably be easier to maintain if there's a way to filter these out from the datasources, I guess?
But doesn't seem that hacky, or I'm not sure if creating a method to see if a node is a browser side query would feel less hacky.